pkg/cli/image/extract/extract.go

```diff
@@ -375,17 +386,33 @@ func (o *ExtractOptions) Run() error {
 					return fmt.Errorf("unable to connect to image repository %s: %v", from.String(), err)
 				}
 
-				srcManifest, location, err := imagemanifest.FirstManifest(ctx, from.Ref, repo, o.FilterOptions.Include)
+				srcManifest, location, err := retrieveSourceManifest(ctx, from, repo, o)
```

Comment:
What is the reason of moving error handling into the `retrieveSourceManifest` function?. As far as I can see, we are still doing error handling below. Now, on the contrary, we are doing same error handling twice(in retrieveSourceManifest and also in below).

I think, it would be better to stay this as is.

Reply:
It was moved into a function because we need to call the set of code twice, and I wanted to avoid duplicating the code. I see where it would be handled twice in `retrieveSourceManifest`; I'm not sure what you mean by 'also in below'. Would you be able to elaborate on that more please?

This was part of accomplishing when we can't find the current $OS/$ARCH, default to linux/$ARCH and warn the user about what is happening as suggested in https://github.com/openshift/oc/pull/1311#issuecomment-1471799668.